internal/pkg/agent/cmd/install.go

```diff
@@ -221,6 +221,28 @@ func installCmd(streams *cli.IOStreams, cmd *cobra.Command) error {
 
 	var ownership utils.FileOwner
 	cfgFile := paths.ConfigFile()
+	if status == install.Installed {
+		// Uninstall the agent
+		progBar.Describe("Uninstalling current Elastic Agent")
+		args := []string{
+			"uninstall",
+			"--force",
+		}
+		execPath, err := exec.LookPath(paths.BinaryName)
+		if err != nil {
+			return fmt.Errorf("unable to find %s on path: %w", paths.BinaryName, err)
+		}
+		uninstall := exec.Command(execPath, args...)
+		uninstall.Stdout = streams.Out
+		uninstall.Stderr = streams.Err
+		if err := uninstall.Start(); err != nil {
+			return fmt.Errorf("unable to start elastic-agent uninstall: %w", err)
+		}
+		if err := uninstall.Wait(); err != nil {
+			return fmt.Errorf("failed to uninstall elastic-agent: %w", err)
+		}
```

Comment:
You should move this to its own function. Then cleanup the errors to be more generic in the function and then wrap the error output like so:

```
err := performUninstall(...)
if err != nil {
    return fmt.Errorf("failed to uninstall current Elastic Agent: %w", err)
}
```

That ensures that the error back to the user is clear that its failing installing the currently installed Elastic Agent.